scripts/create-k3s-cluster-with-SSL-passthrough.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Create a k3s cluster for KubeStellar deployment
+
+port=9443
+wait=true
+
+display_help() {
+  cat << EOF
+Usage: $0 [--port port] [--nowait][-X]
+
+--port port     map the specified host port to the kind cluster port 443 (default: 9443)
+--nowait        when set to true, the script proceeds without waiting for the nginx ingress patching to complete
+-X              enable verbose execution of the script for debugging
+EOF
+}
+
+while (( $# > 0 )); do
+    case "$1" in
+        (-p|--port)
+            if (( $# > 1 ));
+            then { port="$2"; shift; }
+            else { echo "$0: missing port number" >&2; exit 1; }
+            fi;;
+        (--nowait)
+            wait=false;;
+        (-X)
+            set -x;;
+        (-h|--help)
+            display_help
+            exit 0;;
+        (-*)
+            echo "$0: unknown flag" >&2
+            exit 1;;
+        (*)
+            echo "$0: unknown positional argument" >&2
+            exit 1;;
+    esac
+    shift
+done
+
+echo "Creating a k3s cluster with SSL passthrougn and ${port} port mapping..."
+
+if which kubectl > /dev/null ; then
+    if test -f ~/.kube/config; then
+        if $(kubectl cluster-info 2> /dev/null | grep -qe "control plane"); then
+            echo kubernetes cluster is already up!
+            kubectl cluster-info | /bin/head -3
```

Comment:
On my mac:

```
mspreitz@mjs13 ocm-status-addon % /bin/head
zsh: no such file or directory: /bin/head
```

Reply:
Why not just use `head` ?

Reply:
yes, changed, I was recycling some of Ed code... this is still experimental, I think